src/H5Tconv_bitfield.c

```diff
@@ -105,7 +105,9 @@ H5T__conv_b_b(const H5T_t *src, const H5T_t *dst, H5T_cdata_t *cdata, const H5T_
             }
 
             /* Allocate space for order-reversed source buffer */
-            src_rev = (uint8_t *)H5MM_calloc(src->shared->size);
+            if (conv_ctx->u.conv.cb_struct.func)
```

Comment:
This check was added to a few different conversion functions. The `src_rev` buffer will only be used (and therefore only needs to be allocated/freed) if an application datatype conversion exception function is registered and in use.